AK/Error.cpp

```diff
@@ -22,6 +22,22 @@ Error Error::from_string_view_or_print_error_and_return_errno(StringView string_
 #endif
 }
 
+bool Error::operator==(int code) const
```

Comment:
This feels like a trap waiting to bite us. Is there a way to avoid implementing a general operator== like this?

Reply:
Not as far as I can tell, POSIX errno are - at their core - normal `int`s, and we can't really change that due to Lagom either.

As mentioned earlier, one could try to make them discernible by utilizing `DistinctNumeric`, but that would either be limited to the internal Variant only, or we'd have to break compatibility in a lot of existing code (and at that point, the disadvantages of having to convert raw ints to that distinct type everywhere are larger than the comfort that having a shortcut provides).